CourseProject/src/main/java/com/neoflex/CourseProject/service/ValidationServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.neoflex.CourseProject.service;
+
+import com.neoflex.CourseProject.dto.LoanStatementRequestDto;
+import com.neoflex.CourseProject.exception.ValidationException;
+import com.neoflex.CourseProject.pojo.ValidationProperties;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.Period;
+
+@Service
+@Slf4j
+public class ValidationServiceImpl implements ValidationService {
+
+    private final ValidationProperties validationProperties;
+
+    @Autowired
+    public ValidationServiceImpl(ValidationProperties validationProperties){
+        this.validationProperties = validationProperties;
+    }
+
+    @Override
+    public void resultsOfValidation(LoanStatementRequestDto loanStatementRequestDto){
+        log.info(loanStatementRequestDto.toString());
+        areNamesValid(loanStatementRequestDto.getMiddleName());
+        isEmailValid(loanStatementRequestDto.getEmail());
+        areNamesValid(loanStatementRequestDto.getFirstName());
+        areNamesValid(loanStatementRequestDto.getLastName());
+        isPassportSeriesValid(loanStatementRequestDto.getPassportSeries());
+        isPassportNumberValid(loanStatementRequestDto.getPassportNumber());
+        isAmountValid(loanStatementRequestDto.getAmount());
+        isTermValid(loanStatementRequestDto.getTerm());
+        isAdult(loanStatementRequestDto.getBirthdate());
+        log.info("loanStatementRequestDto is valid");
+   }
+
+    private void isEmailValid(String email){
+       if (!email.matches(validationProperties.getEmailRegex())) {
+           log.info("validation exception email is not valid " + email);
+           throw new ValidationException("mail must be like: example@mail.com");
+       }
+   }
+
+    private void areNamesValid(String name){
+         if (name != null && !name.matches(validationProperties.getNamesRegex())) {
+             log.info("validation exception name is not valid " + name);
+             throw new ValidationException("The name must be more than one and less than thirty-one" +
+                     "characters long and written in Cyrillic" + name);
+         }
+    }
+
+     private void isPassportSeriesValid(String passportSeries){
+        if (!passportSeries.matches(validationProperties.getPassportSeriesRegex())) {
+            log.info("validation exception passport series is not valid " + passportSeries);
+            throw new ValidationException("passport series must consist of for digits");
+        }
+    }
+
+     private void isPassportNumberValid(String passportNumber){
+         if (!passportNumber.matches(validationProperties.getPassportNumberRegex())) {
+             log.info("validation exception passport number is not valid " + passportNumber);
```

Comment:
В идеале писать что-то типа:
```suggestion
             log.info("Номер паспорта: '{}' не прошел проверку - должен состоять из 6 цифр", passportNumber);
```
Чтобы понимать, в какой переменной ошибка

Reply:
Но мой пример тебе тоже не особо подходит, потому что ты проверяешь на конкретный формат, хотя можно было сделать две проверки на то, что серия паспорта состоит только из цифр, а не из символов, и на то, что этих цифр только 6. А сейчас ты теряешь эту проверку ввиду обобщенности

Reply:
исправлено